### ISSUE TITLE
[python] Feature-parity gap analysis

### ISSUE THREAD

**Issue by johnkerl** [single-cell-data/TileDB-SOMA#643]
* Between `main-old` and `main`
* ~Between Python and R -- @mlin good POC for R context~ (now tracked in #839 )

Perhaps best to make a cloud-shared spreadsheet we can put X's  in to check off as done -- ? Whatever works best.

**Comment by maniarathi** [comment]
Splitting this issue into two. This issue is focused on performing the gap analysis between `main` and `main-old` in Python. A separate ticket, #839 , captures the work to perform the gap analysis between the Python and R packages.

**Comment by nguyenv** [comment]
The gap analysis between Python's `main` and `old-main` can be found at [this spreadsheet](https://docs.google.com/spreadsheets/d/1vmS5YGJP2_c3PtnCRveHwK3sk-h4Ahk7uu42ZVk1WaI/edit#gid=0). We compared the branches in two ways:

  1. Running through the example notebooks locally and recording necessary changes that needed to made in order to run under main
  2. Recording all the features listed in the tiledbsoma release notes for old-main and marking whether or not they are still present in main
  
There were some minor convience functions that had not been migrated yet like `add_X_layer` though this has since been added. Though overall, no major gaps were found.
